ui/jquery.ui.mouse.js

```diff
@@ -23,10 +23,110 @@
 //JASPERSOFT #1 #3
 define(function(require) {
 
-	var $ = require("./jquery.ui.widget");
-	var isIE = function () {
-		return navigator.appName === "Microsoft Internet Explorer" || navigator.userAgent.indexOf("Trident/") >= 0;
-	};
+    var $ = require("./jquery.ui.widget");
+
+    var isIE = function () {
+        return navigator.appName === "Microsoft Internet Explorer" || navigator.userAgent.indexOf("Trident/") >= 0;
+    };
+
+    var detectedIEVersion = false;
+
+    var getIEVersion = function () {
+        var version = 0;
+        if (detectedIEVersion !== false) {
+            return detectedIEVersion;
+        }
+
+        if (this.isIE()) {
+            if (navigator.appName === "Netscape") {
+                var ua = navigator.userAgent;
+                var re = new RegExp("Trident/.*rv:([0-9]{1,}[\\.0-9]{0,})");
+                if (re.exec(ua) != null) {
+                    version = parseFloat(RegExp.$1);
+                }
+            } else {
+                var msVersion = navigator.appVersion.split("MSIE")[1];
+                version = parseFloat(msVersion);
+            }
+        }
+
+        detectedIEVersion = version;
+        return version;
+    };
+
+    var isIE11 = function () {
+        return this.getIEVersion() === 11;
+    };
+    var inRect = function (rect, x, y) {
+        return (y >= rect.top && y <= rect.bottom) && (x >= rect.left && x <= rect.right);
+    };
+
+    var hasScroll = function ($element, axis) {
+        var overflow = $element.css("overflow"),
+            overflowAxis;
+
+        if (typeof axis === "undefined" || axis === "y") {
+            overflowAxis = $element.css("overflow-y");
+        } else {
+            overflowAxis = $element.css("overflow-x");
+        }
+
+        var bShouldScroll = $element.get(0).scrollHeight > $element.innerHeight();
+
+        var bAllowedScroll = (overflow === "auto" || overflow === "visible") ||
+            (overflowAxis === "auto" || overflowAxis === "visible");
+
+        var bOverrideScroll = overflow === "scroll" || overflowAxis === "scroll";
+
+        return (bShouldScroll && bAllowedScroll) || bOverrideScroll;
+    };
+
+    var inScrollRange = function (event) {
+
+        var
+            scrollSize = 18,
```

Comment:
Why you hardcoded scrollSize?
I'm pretty sure that it's possible to change the scrollbar thickness via settings or even via css.

Reply:
Do you mean "-webkit-scrollbar" family rules ? They are not supported by IE:
https://css-tricks.com/almanac/properties/s/scrollbar/

Reply:
right but it's possible to change scrollbar width system-wide in windows:
[how-to-adjust-the-width-of-the-scrollbar](https://answers.microsoft.com/en-us/windows/forum/windows_7-desktop/how-to-adjust-the-width-of-the-scrollbar-in/b45bf22b-4882-45e1-9b3c-5051da0dc631?auth=1)

Reply:
I see, let me try to find out the scroll size in this case, should be similar to the same as before